src/mlpack/methods/ann/layer/threshold_fn.hpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * @file methods/ann/layer/threshold_fn.hpp
+ * @author Shubham Agrawal
+ *
+ * Definition of the Threshold activation function
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_THRESHOLD_HPP
+#define MLPACK_METHODS_ANN_LAYER_THRESHOLD_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The Threshold activation function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     x & : x > threshold \\
+ *     value & : x \le threshold
+ *   \end{array}
+ * \right.
+ * @f}
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class Threshold
+{
+ public:
+  /**
+   * Create the Threshold object using the specified parameters.
+   * The non zero gradient region can be adjusted by specifying the parameter
+   * threshold. Default (threshold = 0.0)
+   * The value when x < threshold can be adjusted by specifying the parameter
+   * value. Default (value = 0.0)
+   *
+   * @param threshold Non zero gradient region threshold
+   * @param value Value when x < threshold
```

Comment:
```suggestion
   * @param threshold Non zero gradient region threshold.
   * @param value Value when x < threshold.
```

See https://github.com/mlpack/mlpack/wiki/DesignGuidelines#comments for more information.